### ISSUE TITLE
Show file/directory in default native OS' file manager

### ISSUE THREAD

**Issue by Anoesj** [WICG/native-file-system#156]
Imagine a scenario where an app (e.g., a web-based code editor) asks the user to select a directory to open using Native File System. This app sometimes saves (new) files into this directory programmatically, and afterwards it would like to notify the user that a file has been created/updated and grant them the possibility to open this file in their OS' default native file manager (Finder, Windows Explorer, etc). For example, by showing a toast notification saying: 
> File saved to disk. <a href="#">Show saved file</a>.

I couldn't find anything in the Native File System spec that would allow me to do that, but it would be nice to have this possibility!

**Comment by mkruisselbrink** [comment]
Agree that this seems like a nice feature. I think it would address some of the use cases for which people have requested access to the full path, while having different (fewer?) security/privacy issues.

Although a web-api that can trigger the OS to do something does of course still have potential security implications, so would definitely have to be careful about what exactly we allow.

**Comment by kepta** [comment]
Since showing the full path is out of scope (https://github.com/WICG/file-system-access/issues/178), I am also facing hurdle in providing a good user experience to the users of my app. Once a directory handle is saved in browsers storage, the user has no way of knowing where it is in their computer. To give an example, imagine using an online text editor and giving it permission to some random folder buried deep in your computer and now you have no idea on how to find it, except for searching by directory name. 


Here is how I imagine this api to be:
```js
window.openFile(dirHandle); // returns void
```

To avoid the misuse and popup problems, the API will only open the native file explorer when `dirHandle` has permissions.

> Although a web-api that can trigger the OS to do something does of course still have potential security implications, so would definitely have to be careful about what exactly we allow.

I think an api which allows opening of the saved file handle in the OS's native file explorer will improve the user experience tremendously. I am not seeing how this is different from chrome showing the native file picker, in both cases the result is entirely opaque to javascript.

**Comment by ftreesmilo** [comment]
I need something like this.  Without full path support, I can't fix issues I'm getting reports from users not remembering where the folder they granted me access to was.  This is a large UX issue that I need fixed.

**Comment by mkruisselbrink** [comment]
I agree that this would be a nice feature to have. Unfortunately I don't think we'll get to this in the near future, but it's definitely on our backlog of things we would like to add.

**Comment by mh-cbon** [comment]
I need that feature too, absolute file path. 

I am having an app in **secure context**, the user has to choose a folder, its absolute path is then forwarded to the backend for "the business". The backend does not care of a file handle. 

I was thrilled to replace my odious textarea to input the directory path with that new API : / I am less thrilled to re invent the wheel = ) Hope you can figure out a way to make it really cool.

As a site note, i am really not interested in running electron.... Also `<input id="myFile" type="file" webkitdirectory />` does not seem to be a good solution, when i select a directory it seems to import the whole tree within the browser.

**Comment by jespertheend** [comment]
@mh-cbon This issue is about opening the file explorer. I think the request you're looking for is #145

**Comment by jimmywarting** [comment]
This deserves a little bump +1 (again)

**Comment by IanBellomy** [comment]
In addition to the concrete issue @ftreesmilo notes, I think the absence of this ability can end up being grating across different user perspectives on directory / file selection: 

... _Why can't I access what I just gave you and which is still mine?_

... _I just input all these (directory) selections, but you lost my choices?_

... _Why can't I use the link I made?_ 

... _I just set _this_ thing to be _that_ thing, why is it no longer that thing?_ 

... _Why can't I just go back?_

... _Why can't I use this file/directory like a file / directory anymore?_

**Comment by jimmywarting** [comment]
> Here is how I imagine this api to be:
> ```js
> openFile(dirHandle); // returns void
> ```
> To avoid the misuse and popup problems, the API will only open the native file explorer when dirHandle has permissions.

`openFile` is kind of wrong term here if it should only _show_ the file. open can mean: open the file in the default application, making it executable. better would be:
```js
// show file or directory handle in window file explorer.exe or mac's finder.app
showFileSystemHandle(handle)
```

**Comment by noobSam1122** [comment]
When a file handle was granted, the user should have the ability to open it (in explorer), have no idea why this simple thing still not implemented yet..
